app/ssttpcalculator/CalculatorService.scala

```diff
@@ -230,8 +231,8 @@ class CalculatorService @Inject() (
     val sortedDebits: Seq[DebitInput] = debits.sortBy(_.dueDate)
 
       def processDebits(amount: BigDecimal, debits: Seq[DebitInput]): BigDecimal = {
-        debits match {
-          case debit :: Nil => calculateAmount(amount, debit)._1 * calculateDays(debit) * currentDailyRate
+        debits.toList match {
```

Comment:
Do we need recursion for this might be a easier way to write it...Also good pratice to add that @tail.annocate thing to make sure it  checks it will not blow up

Reply:
tried to rewrite it as tail-recursive but didn't pass tests, will leave as is seeing as it's been serving in production well